File: apps/ios/Pebbles/Features/Path/Components/RippleStrokeColor.swift
import SwiftUI

/// Logical color slot for a Ripple stroke. The view maps each tone
/// to a theme-aware `Color` asset.
enum RippleStrokeTone: Equatable {
    case `default`   // outside the user's current level
    case active      // within level, user created a pebble today
    case inactive    // within level, user has NOT created a pebble today
}

/// Pure mapping from `(strokeId, level, activeToday)` to a `RippleStrokeTone`.
/// Encodes the truth table from issue #442 verbatim:
///   - strokeId > level                      → .default
///   - strokeId <= level &&  activeToday     → .active
///   - strokeId <= level && !activeToday     → .inactive
func rippleStrokeTone(strokeId: Int, level: Int, activeToday: Bool) -> RippleStrokeTone {
    if strokeId > level { return .default }
    return activeToday ? .active : .inactive
}

extension RippleStrokeTone {
    /// Resolved theme-aware color for this tone.
    var color: Color {
        switch self {
        case .default:  return .rippleDefault
        case .active:   return .rippleActive
        case .inactive: return .rippleInactive
        }
    }
}

File: apps/android/app/src/main/kotlin/app/pbbls/android/features/shared/ripples/RippleStrokeColor.kt
package app.pbbls.android.features.shared.ripples

import androidx.compose.ui.graphics.Color
import app.pbbls.android.theme.PebblesColors

/**
 * Logical color slot for a Ripple stroke — ports iOS `RippleStrokeColor.swift`.
 * [color] resolves a tone against the active palette bundle (a pure function,
 * not a composable, so DrawScope lambdas can call it).
 */
enum class RippleStrokeTone {
    DEFAULT, // outside the user's current level
    ACTIVE, // within level, user created a pebble today
    INACTIVE, // within level, user has NOT created a pebble today
}

/**
 * Pure mapping from `(strokeId, level, activeToday)` to a [RippleStrokeTone].
 * Encodes the truth table from issue #442 verbatim:
 *   - strokeId > level                  → DEFAULT
 *   - strokeId <= level &&  activeToday → ACTIVE
 *   - strokeId <= level && !activeToday → INACTIVE
 */
fun rippleStrokeTone(
    strokeId: Int,
    level: Int,
    activeToday: Boolean,
): RippleStrokeTone =
    when {
        strokeId > level -> RippleStrokeTone.DEFAULT
        activeToday -> RippleStrokeTone.ACTIVE
        else -> RippleStrokeTone.INACTIVE
    }

/** Resolved theme-aware color for this tone. */
fun RippleStrokeTone.color(colors: PebblesColors): Color =
    when (this) {
        RippleStrokeTone.DEFAULT -> colors.system.muted
        RippleStrokeTone.ACTIVE -> colors.accent.primary
        RippleStrokeTone.INACTIVE -> colors.system.secondary
    }
